packages/components/src/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { useTabList } from '@react-aria/tabs';
+import {
+  Box,
+  ThemeExtensionsWithParts,
+  useComponentStyles,
+} from '@marigold/system';
+import { useTabListState } from '@react-stately/tabs';
+import { useRef } from 'react';
+import { AriaTabListProps } from '@react-types/tabs';
+
+import { Tab } from './Tab';
+import { TabPanel } from './TabPanel';
+import { TabContext } from './Context';
+import { Item } from '@react-stately/collections';
+
+export interface TabsThemeExtension
+  extends ThemeExtensionsWithParts<'Tabs', ['tabs', 'tab', 'tabPanel']> {}
+export interface TabsProps
+  extends Omit<AriaTabListProps<object>, 'orientation'> {
+  gap?: number;
+  size?: 'small' | 'medium' | 'large';
+}
+export const Tabs = ({ gap = 1, size = 'medium', ...props }: TabsProps) => {
+  const state = useTabListState(props);
+  const ref = useRef(null);
+  const { tabListProps } = useTabList(props, state, ref);
+  const styles = useComponentStyles(
+    'Tabs',
+    { size },
+    { parts: ['tabs', 'tab', 'tabPanel'] }
+  );
+  return (
+    <TabContext.Provider value={{ styles }}>
+      <Box>
+        <Box
+          css={styles.tabs}
+          __baseCSS={{ display: 'flex', gap: `${gap}rem` }}
+          {...tabListProps}
+          ref={ref}
+        >
+          {[...state.collection].map(item => {
+            return <Tab key={item.key} item={item} state={state} />;
+          })}
+        </Box>
+        <TabPanel key={state.selectedItem?.key} state={state} />
```

Comment:
Can you help me? What does this do?

Reply:
As you know when we click on `Tab` we will get the content "`TabPanel`" which is connected to this `Tab` we clicked.
Every `Tab` has a **key**, so when you click on `Tab` , the **key** is set into `state.selectedItem` and we use this to show the active `TabPanel`